### PR TITLE
Account settings parent info

### PR DESCRIPTION
**Pull request by bethanyaconnor**
To test go to http://localhost-studio.code.org:3000/users/edit?enableExperiments=parent-email-preference

This PR adds a controller that interfaces between the account setting page and the parent email modal as well as adding a section to the account setting page to add a parent email. This section should only be visible on a student account.

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/ui/accounts/AddParentEmailController.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+   */
+  constructor({form, link, displayedParentEmail}) {
+    this.form = form;
+    this.displayedParentEmail = displayedParentEmail;
+    link.click(this.showAddParentEmailModal);
+  }
+
+  showAddParentEmailModal = () => {
+    if (this.mountPoint) {
+      console.log('already have a mount point');
```
Comment (daynew) [inline apps/src/lib/ui/accounts/AddParentEmailController.js]: Is this a debug statement you mean to remove before committing?

**Comment by bethanyaconnor** [comment]
Adhoc: https://adhoc-account-settings-parent-info-studio.cdn-code.org/courses?enableExperiments=parent-email-preference

File: dashboard/app/controllers/registrations_controller.rb [hunk 2]
```diff
@@ -138,6 +138,24 @@ def destroy
...
   end
 
+  def parent_email_params
+    params.
+      require(:user).
+      tap do |user|
+        user[:parent_email_preference_email] = user[:parent_email]
+        user[:parent_email_preference_opt_in_required] = '1'
+        user[:parent_email_preference_request_ip] = request.ip
+        user[:parent_email_preference_source] = EmailPreference::ACCOUNT_SIGN_UP
```
Comment (daynew) [inline dashboard/app/controllers/registrations_controller.rb]: Do we need a new SOURCE_TYPE for changes made on the edit account page?

**Review by daynew — COMMENTED** [review]
Looks good!

File: apps/src/lib/ui/accounts/AddParentEmailController.js (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/lib/ui/accounts/AddParentEmailController.js]: I'll remove

File: dashboard/app/controllers/registrations_controller.rb (hunk 2 above)
Comment (bethanyaconnor) [reply dashboard/app/controllers/registrations_controller.rb]: Oh good call!

**Pull request closed without merging**